grobid-core/src/main/java/org/grobid/core/jni/JEPThreadPool.java

```diff
@@ -55,7 +55,54 @@ private JEPThreadPool() {
         //executor = Executors.newFixedThreadPool(POOL_SIZE); 
         executor = Executors.newSingleThreadExecutor();
         // each of these threads is associated to a JEP instance
-        jepInstances = new HashMap<>();
+        jepInstances = new ConcurrentHashMap<>();
```

Comment:
[HashMap](https://docs.oracle.com/javase/8/docs/api/java/util/HashMap.html) is not thread-safe. Using [ConcurrentHashMap](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html) instead. Although I would recommend using [ThreadLocal](https://docs.oracle.com/javase/8/docs/api/java/lang/ThreadLocal.html). (See [comment](https://github.com/kermitt2/grobid/pull/369/files#r296194872), [comment](https://github.com/kermitt2/grobid/pull/369/files#r296184959))

Reply:
yes very good point ! This might cause errors when 2 threads try to allocate concurrently JEP instances.